discovery_engine/lib/src/ffi/types/document/feedback.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:ffi' show Pointer;
+
+import 'package:xayn_discovery_engine/src/domain/models/document.dart'
+    show DocumentFeedback;
+import 'package:xayn_discovery_engine/src/ffi/genesis.ffigen.dart'
+    show RustUserReaction;
+import 'package:xayn_discovery_engine/src/ffi/load_lib.dart' show ffi;
+
+extension DocumentFeedbackFfi on DocumentFeedback {
+  void writeNative(final Pointer<RustUserReaction> place) {
+    var asInt = 128;
+    switch (this) {
+      case DocumentFeedback.neutral:
+        asInt = 0;
```

Comment:
Can we use `serde_repr` like in xayn-ai so we don't have to keep the values in sync?

Reply:
`serde_repr` is mainly used due to wasm-bindgen using `Serde`.

In dart we can try to abuse the json code genration again, but that is tricky due to:

- us relying on implementation details of the json library
- relying on the annotations in dart and rust being the same
- it not generating the int mapping if the enum is not used in another json annotated struct.

Alternatively we can use `cbindgen` to spider into `core`  and
then have `DocumentFeedbackFfi` constants and maybe?? we can put that into
the json libraries annotations?

In rust we could use `serde_repr` + `serde_json`, or `as` cast + bounds checked transmute.